### PR TITLE
Download file handler grpc gateway WIP

### LINKED ISSUE #313: upload/download http handlers to grpc middleware

### PR DESCRIPTION
**Pull request by senpathi**
resolves #313

### CODE REVIEW

File: code/go/0chain.net/blobbercore/handler/handler.go [hunk 1]
```diff
@@ -125,6 +125,18 @@ func setupHandlerContext(ctx context.Context, r *http.Request) context.Context {
 	return ctx
 }
 
+func setupGRPCMuxHandlerContext(ctx context.Context, r *http.Request, pathParams map[string]string) context.Context {
```
Comment (kushthedude) [inline code/go/0chain.net/blobbercore/handler/handler.go:128]: Why? We already have a similar function, look at the functions at `grpc_handler.go`

File: code/go/0chain.net/blobbercore/handler/grpc_helpers.go [hunk 2]
```diff
@@ -83,3 +83,10 @@ func CustomMatcher(key string) (string, bool) {
 		return runtime.DefaultHeaderMatcher(key)
 	}
 }
+
+func uploadHandler(w http.ResponseWriter, r *http.Request, pathParams map[string]string) {
```
Comment (kushthedude) [inline code/go/0chain.net/blobbercore/handler/grpc_helpers.go:87]: No need for this actually

File: code/go/0chain.net/blobbercore/handler/grpc_helpers.go [hunk 3]
```diff
@@ -18,14 +18,14 @@ type gRPCHeaderMetadata struct {
...
 func registerGRPCServices(r *mux.Router, server *grpc.Server) {
 	blobberService := newGRPCBlobberService()
 	r.Use(Middleware2("ds"))
 	grpcGatewayHandler := runtime.NewServeMux(
-								runtime.WithIncomingHeaderMatcher(CustomMatcher),
-							)
+		runtime.WithIncomingHeaderMatcher(CustomMatcher),
+	)
 
+	_ = grpcGatewayHandler.HandlePath("POST", `/v1/file/upload/{allocation}`, uploadHandler)
```
Comment (kushthedude) [inline code/go/0chain.net/blobbercore/handler/grpc_helpers.go:28]: Can we not re-use the already implemented HTTP handler in the `handler.go` file?

**Review by kushthedude — CHANGES_REQUESTED** [review]

File: code/go/0chain.net/blobbercore/handler/handler.go (hunk 1 above)
Comment (senpathi) [reply code/go/0chain.net/blobbercore/handler/handler.go:128]: Yes, this to get path params or we need to change that function to get map[string]string

Comment (kushthedude) [reply code/go/0chain.net/blobbercore/handler/handler.go:128]: We already have function to do this see this line in the `grpc_handlers.go` `setupGrpcHandlerContext(ctx, getGRPCMetaDataFromCtx(ctx))`

File: code/go/0chain.net/blobbercore/handler/grpc_helpers.go (hunk 2 above)
Comment (senpathi) [reply code/go/0chain.net/blobbercore/handler/grpc_helpers.go:87]: where should this handler be implemented?

File: code/go/0chain.net/blobbercore/handler/grpc_helpers.go (hunk 3 above)
Comment (senpathi) [reply code/go/0chain.net/blobbercore/handler/grpc_helpers.go:28]: Yes, we can rewrite StorageHandler Methods for it

File: code/go/0chain.net/blobbercore/handler/grpc_helpers.go (hunk 2 above)
Comment (kushthedude) [reply code/go/0chain.net/blobbercore/handler/grpc_helpers.go:87]: No actually we already have this implemented, we can just use that. See the `UploadHandler` in the handler.go file.

**Review by kushthedude — APPROVED** [review]

**Pull request merged**